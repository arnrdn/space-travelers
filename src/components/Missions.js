import React from 'react';

const Missions = () => <div>Draft</div>;

export default Missions;
