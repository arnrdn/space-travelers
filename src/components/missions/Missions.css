* {
  margin: 0;
  padding: 0;
}

ul {
  list-style: none;
}

.missions-cols,
.mission {
  width: 90%;
  margin: 0 auto;
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 1fr 4fr 1fr 1fr;
}

.mission:nth-of-type(odd) {
  background-color: #f2f2f2;
}

.mission-data {
  box-sizing: border-box;
  padding: 0.8rem 0.4rem 1.75rem 0.4rem;
  border: 1px solid #dee2e6;
}

.mission .mission-data:first-of-type {
  font-weight: 700;
  font-size: 1.25rem;
}

.mission-col {
  box-sizing: border-box;
  border: 1px solid #dee2e6;
  padding: 0.5rem 0.35rem;
  font-size: 1.25rem;
  font-weight: 700;
}
