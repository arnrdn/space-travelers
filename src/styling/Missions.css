* {
  margin: 0;
  padding: 0;
}

ul {
  list-style: none;
}

.missions-container {
  margin: 4rem auto;
}

.missions-cols,
.mission {
  width: 90%;
  margin: 0 auto;
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 1fr 4fr 1fr 1fr;
}

.mission:nth-of-type(odd) {
  background-color: #f2f2f2;
}

.mission-data {
  box-sizing: border-box;
  padding: 0.8rem 0.4rem 1.75rem 0.4rem;
  border: 1px solid #dee2e6;
}

.mission .mission-data:first-of-type {
  font-weight: 700;
  font-size: 1.25rem;
}

.mission .mission-data:nth-of-type(3),
.mission .mission-data:nth-of-type(4) {
  display: flex;
  justify-content: center;
  align-items: center;
}

.btn-m {
  box-sizing: border-box;
}

.mission-col {
  box-sizing: border-box;
  border: 1px solid #dee2e6;
  padding: 0.5rem 0.35rem;
  font-size: 1.25rem;
  font-weight: 700;
}

.mission-active {
  color: #fff;
  font-weight: 600;
  padding: 0.5rem;
  border-radius: 5px;
}

.non-m {
  background-color: #6d757d;
}

.m {
  background-color: #18a2b8;
}

.mission-data button {
  background-color: transparent;
  padding: 0.5rem;
  border-radius: 5px;
  font-size: 1rem;
  cursor: pointer;
}

.mission-join {
  border: 1px solid #343a40;
  color: #343a40;
}

.mission-leave {
  border: 1px solid #de3e4d;
  color: #de3e4d;
}
