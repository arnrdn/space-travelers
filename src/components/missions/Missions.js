import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchMissionsApi } from '../../redux/missions/missions';
import Mission from './Mission';

const Missions = () => {
  const missionsData = useSelector((state) => state.missionsReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMissionsApi());
  }, []);

  const missions = missionsData.map((item) => (
    <li className="mission" key={item.id}>
      <Mission mission={item} />
    </li>
  ));

  return (
    <div className="missions-container">
      <ul className="missions-cols">
        <li className="mission-col">
          Mission
        </li>
        <li className="mission-col">
          Description
        </li>
        <li className="mission-col">
          Status
        </li>
        <li className="mission-col">
          /
        </li>
      </ul>
      <ul className="missions">
        {missions}
      </ul>
    </div>
  );
};

export default Missions;
