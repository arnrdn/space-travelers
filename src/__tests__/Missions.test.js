import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Missions from '../components/missions/Missions';

describe('MyProfile tests', () => {
  test('should render', () => {
    const missions = render(
      <Provider store={store}>
        <BrowserRouter>
          <Missions />
        </BrowserRouter>
        ,
      </Provider>,
    );

    expect(missions).toMatchSnapshot();
  });
});
