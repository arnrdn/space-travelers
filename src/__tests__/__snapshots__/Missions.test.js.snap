// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="missions-container"
      >
        <ul
          class="missions-cols"
        >
          <li
            class="mission-col"
          >
            Mission
          </li>
          <li
            class="mission-col"
          >
            Description
          </li>
          <li
            class="mission-col"
          >
            Status
          </li>
          <li
            class="mission-col"
          >
            /
          </li>
        </ul>
        <ul
          class="missions"
        />
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="missions-container"
    >
      <ul
        class="missions-cols"
      >
        <li
          class="mission-col"
        >
          Mission
        </li>
        <li
          class="mission-col"
        >
          Description
        </li>
        <li
          class="mission-col"
        >
          Status
        </li>
        <li
          class="mission-col"
        >
          /
        </li>
      </ul>
      <ul
        class="missions"
      />
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
